utils/common/fixedallocator.h

```diff
@@ -89,10 +97,46 @@ class FixedAllocator
     uint64_t currentlyStored;
     bool tmpSpace;
     uint8_t* nextAlloc;
+    bool useLock;
+    std::atomic<bool> lock;
 };
 
+inline void* FixedAllocator::allocate()
+{
+    void* ret;
+
+    if (useLock)
+        getSpinlock(lock);
```

Comment:
This spinlock potentially could be very painfull.

Reply:
True.  When it does allocate add'l mem, threads could pile up there.  For the typical case when it's not allocating, much less risk.  I found it was better to copy *allocator instances than to have threads share one, but decided to leave the mods in there b/c they're harmless with the defaults.